src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -829,6 +829,46 @@ ov::pass::NopSliceBeforeGatherElements::NopSliceBeforeGatherElements() {
     register_matcher(m, matcher_pass_callback);
 }
 
+ov::pass::PrepareShapeOpsForEliminationAroundBE::PrepareShapeOpsForEliminationAroundBE() {
+    MATCHER_SCOPE(PrepareShapeOpsForEliminationAroundBE);
+    auto first_label = pattern::wrap_type<op::v1::Reshape, op::v0::Squeeze>(pattern::rank_equals(0));
+    auto other_input_label = pattern::any_input(pattern::rank_equals(0));
+    auto binary_op_label = pattern::wrap_type<op::util::BinaryElementwiseArithmetic,
+                                              op::util::BinaryElementwiseComparison,
+                                              op::util::BinaryElementwiseLogical>({first_label, other_input_label},
+                                                                                  pattern::consumers_count(1));
+    auto second_label = pattern::wrap_type<op::v1::Reshape, op::v0::Unsqueeze>({binary_op_label, pattern::any_input()},
+                                                                               pattern::rank_equals(1));
+
+    ov::matcher_pass_callback matcher_pass_callback = [=](pattern::Matcher& m) {
+        const auto& pattern_to_node = m.get_pattern_map();
+
+        auto second_node = pattern_to_node.at(second_label);
+        auto binary = pattern_to_node.at(binary_op_label);
+
+        auto lhs_node =
+            ov::op::util::clone_try_fold(second_node, {binary->input_value(0), second_node->input_value(1)});
+        auto rhs_node =
+            ov::op::util::clone_try_fold(second_node, {binary->input_value(1), second_node->input_value(1)});
+
+        register_new_node(lhs_node);
+        register_new_node(rhs_node);
+
+        binary->input(0).replace_source_output(lhs_node->output(0));
+        binary->input(1).replace_source_output(rhs_node->output(0));
+        binary->validate_and_infer_types();
```

Comment:
maybe it's better to clone it?
binary->clone_with_new_inputs?

Reply:
We have no need to create new node here since we know that patents have this consumer exclusively.